packages/gatsby-source-mongodb/src/__tests__/stringify-object-ids.js

```diff
@@ -0,0 +1,168 @@
+const stringifyObjectIds = require(`../stringify-object-ids`)
+const ObjectID = require(`mongodb`).ObjectID
+
+test(`it is a function`, () => {
+  expect(stringifyObjectIds).toEqual(expect.any(Function))
+})
+
+test(`ObjectIDs are instances of mongodb.ObjectID and not strings`, () => {
+  const objectId = new ObjectID(`5c6d85cff2bc0b7a5da61124`)
+  expect(objectId instanceof ObjectID).toBeTruthy()
+  expect(typeof objectId == `string`).toBeFalsy()
+})
+
+test(`ObjectIDs are not Equal to Strings with the same value`, () => {
+  const objectId = new ObjectID(`5c6d85cff2bc0b7a5da61124`)
+  const str = `5c6d85cff2bc0b7a5da61124`
+  expect(objectId).not.toEqual(str)
+  expect(objectId)
+})
+
+test(`it returns stringified object id when passed one`, () => {
+  const objectId = new ObjectID(`5c6d85cff2bc0b7a5da61124`)
+  expect(stringifyObjectIds(objectId)).toEqual(`5c6d85cff2bc0b7a5da61124`)
+})
+
+test(`when passed an array of ObjectIDs it returns an array of strings`, () => {
+  const str1 = `5c6d85cff2bc0b7a5da61124`,
+    str2 = `5c6d85fdf2bc0b7a5da6112a`,
+    str3 = `5c6d863df2bc0b7a5da61132`,
+    obj1 = new ObjectID(str1),
+    obj2 = new ObjectID(str2),
+    obj3 = new ObjectID(str3),
+    arr = [obj1, obj2, obj3],
+    res = [str1, str2, str3],
+    stringifiedArr = stringifyObjectIds(arr)
+
+  expect(stringifiedArr[0]).toEqual(res[0])
+  expect(stringifiedArr[1]).toEqual(res[1])
+  expect(stringifiedArr[2]).toEqual(res[2])
+})
+
+test(`when passed an array of object literals with ObjectIDs it returns an array of object literals with strings`, () => {
+  const str1 = `5c6d85cff2bc0b7a5da61124`,
+    str2 = `5c6d85fdf2bc0b7a5da6112a`,
+    str3 = `5c6d863df2bc0b7a5da61132`,
+    obj1 = new ObjectID(str1),
+    obj2 = new ObjectID(str2),
+    obj3 = new ObjectID(str3),
+    obj = [
+      {
+        key: `one`,
+        id: obj1,
+      },
+      {
+        key: `two`,
+        id: obj2,
+      },
+      {
+        key: `three`,
+        id: obj3,
+      },
+    ],
+    res = [
+      {
+        key: `one`,
+        id: str1,
+      },
+      {
+        key: `two`,
+        id: str2,
+      },
+      {
+        key: `three`,
+        id: str3,
+      },
+    ],
+    stringifiedObj = stringifyObjectIds(obj)
+
+  expect(stringifiedObj[0].id).toEqual(res[0].id)
+  expect(stringifiedObj[1].id).toEqual(res[1].id)
+  expect(stringifiedObj[2].id).toEqual(res[2].id)
+})
+
+test(`when passed a deeply nested, complex object literal, it returns all stringified ObjectIds`, () => {
+  const str1 = `5c6d85cff2bc0b7a5da61124`,
```

Comment:
This one might be a little more challenging to write in a cleaner way--so let me know if you'd like an assist!

Reply:
I'm not sure how cleaner I got it:

    const ids = [
      new ObjectID().toHexString(),
      new ObjectID().toHexString(),
      new ObjectID().toHexString(),
      new ObjectID().toHexString(),
      new ObjectID().toHexString(),
      new ObjectID().toHexString(),
    ]
    const complexNested = {
      id: new ObjectID(ids[0]),
      connections: [
        {
          id: new ObjectID(ids[1]),
        },
        {
          ids: [new ObjectID(ids[2]), new ObjectID(ids[3])],
        },
      ],
      name: {
        first: `sally`,
        last: {
          maiden: {
            val: `smith`,
            id: new ObjectID(ids[4]),
          },
          current: {
            val: `wilson`,
            id: new ObjectID(ids[5]),
          },
          former: {
            val: `smith`,
            id: new ObjectID(ids[4]),
          },
        },
      },
    }
    const expectedResult = {
      id: ids[0],
      connections: [
        {
          id: ids[1],
        },
        {
          ids: [ids[2], ids[3]],
        },
      ],
      name: {
        first: `sally`,
        last: {
          maiden: {
            val: `smith`,
            id: ids[4],
          },
          current: {
            val: `wilson`,
            id: ids[5],
          },
          former: {
            val: `smith`,
            id: ids[4],
          },
        },
      },
    }

    expect(stringifyObjectIds(complexNested)).toEqual(expectedResult)
    expect(stringifyObjectIds(complexNested)).not.toEqual(complexNested)